src/vm/wren_core.c

```diff
@@ -472,6 +482,11 @@ DEF_PRIMITIVE(fiber_yield1)
 
 DEF_PRIMITIVE(fn_new)
 {
+  RETURN_ERROR("Fn metaclass does not implement 'new()'.");
+}
```

Comment:
> From the compiler's perspective it sees a class without any constructors and creates a default constructor for it anyways.

Ah, right. Because primitives are injected in the class after the fact unlike foreign methods.

One option would be to put the fake constructors that abort in the .wren code (using `Fiber.abort()`) instead of as a primitive.

It does feel like we may need a more general-purpose solution, though. For example, `Sequence` is basically an abstract base class. It's designed to be extended but not constructed. There isn't really any good way to express that right now.

This touches on some areas I've been thinking a lot about lately (including stuff like mixins), but I don't know if I have any good solutions yet.